wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift

```diff
@@ -42,43 +67,35 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
     private var didInitializeMLS = false
     private let databasePath: String
 
-    public convenience init(coreCryptoConfiguration config: CoreCryptoConfiguration) throws {
+    public convenience init(coreCryptoConfiguration config: CoreCryptoConfiguration) async throws {
         guard let clientID = config.clientIDBytes else {
             throw CoreCryptoSetupFailure.failedToGetClientIDBytes
         }
 
-        let coreCrypto = try CoreCrypto(
-            path: config.path,
-            key: config.key,
-            clientId: clientID,
-            ciphersuites: [defaultCipherSuite.rawValue]
-        )
+        let coreCrypto = try await coreCryptoNew(path: config.path, key: config.key, clientId: clientID, ciphersuites: [CiphersuiteName.default.rawValue], nbKeyPackage: nil)
 
         self.init(coreCrypto: coreCrypto, databasePath: config.path)
         didInitializeMLS = true
     }
 
-    public convenience init(path: String, key: String) throws {
-        let coreCrypto = try CoreCrypto.deferredInit(
-            path: path,
-            key: key,
-            ciphersuites: [defaultCipherSuite.rawValue]
-        )
+    public convenience init(path: String, key: String) async throws {
```

Comment:
Would making the CiphersuiteName be part of the initialiser to make it configurable?

Reply:
Hmm actually the ciphersuite is irrelevant in this case we are only initialising proteus here and it doesn't care about the ciphersuite.

Reply:
I'll add comment